### DISCUSSION TITLE
How do I suppress the warning for "undefined"?

### DISCUSSION THREAD

**Discussion by Aster89** [haskell/haskell-language-server#4045] [Q&A]
I make a massive use of `undefined` because it helps me filling the holes where I don't know yet what to write; then I often query its type (which is deduced by the surrounding context) and conclulde "oh, yes, then this needs to be a `x $ y z`).

Having `undefined` marked in yellow (or whatever, I use HLS on Vim via YouCompleteMe, for what is worth) it's really distracting.

**Comment by jhrcek** [comment]
Can you share more details about your project?

Afaik there's no setting at the level of haskell language server that would allow disabling this warning.
But I know that for example relude library exports a version of undefined with which a warning is associated:
https://hackage.haskell.org/package/relude-1.2.1.0/docs/src/Relude.Debug.html#undefined

Are you perhaps using relude or similar library?
In general I suspect that you might be dealing with https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/pragmas.html?highlight=pragmas#pragma-WARNING

**Reply by Aster89** [reply]
I don't think I'm using that `Relude` thing...

Here's more information that might be useful:

- hovering on `undefined` shows
  > ```haskell
  > undefined :: forall a. HasCallStack => a
  > ```
  >
  > *Defined in ‘GHC.Err’* *(base-4.17.2.1)*
  >
  > * * *
  >
  > ```haskell
  > _ :: IO Report
  > ```
  > * * *
  >
  > ```haskell
  > _ :: forall a. HasCallStack => a
  > ```
- Compiler: GHC 9.4.8
- Language server: HLS 2.5.0.0

What other information can I provide?

**Comment by michaelpj** [comment]
What is the diagnostic that you actually get?

**Reply by Aster89** [reply]
```
  ✲ Name:        Anti-pattern: unsafe functions
  ✲ Description: Usage of unsafe functions breaks referential transparency
  ✲ Severity:    Error
  ✲ Category:    #Unsafe #AntiPattern
 Possible solutions:
   - Remove 'undefined' or at least replace with 'error' to give better error messages
   - Replace 'unsafeCoerce' with 'coerce'
   - Rewrite the code to avoid using 'unsafePerformIO' and other unsafe IO functions
  [STAN-0212]
```

**Reply by michaelpj** [reply]
This is a stan diagnostic. The stan plugin is off in more recent versions of HLS, or you can turn it off in your editor's config.

**Comment by jhrcek** [comment]
Ah, I didn't think of that, because I never saw stan working :)
There should be two ways to disable stan warning:
1. disable stan plugin at the level of haskell-language-server configuration
2. disable this particular diagnostic at the level of stan (the stan itself can be configured via [config files](https://github.com/kowainik/stan?tab=readme-ov-file#toml-configurations)). Though you should be aware that the way that stan plugin is loading config files has been modified/fixed only recently - after the official hls 2.6.0.0 release (cf [PR](https://github.com/haskell/haskell-language-server/pull/4023))
